### PR TITLE
Handle inventory types where Automation Hub collection names differ

### PR DESCRIPTION
**Pull request by AlanCoding**
Some collections will be _moving_, which is #10323

Yet, other collections will _change name_, which this is intended to address.

### CODE REVIEW

File: awx/main/utils/licensing.py [hunk 1]
```diff
@@ -379,10 +377,10 @@ def validate(self):
             return attrs
         attrs['valid_key'] = True
 
-        if Host:
```
Comment (AlanCoding) [inline awx/main/utils/licensing.py:382]: `if Host:`, but the `Host` here was the class `Host`. This should always be True, unless I'm crazy.

File: awx/api/generics.py [hunk 2]
```diff
@@ -226,7 +224,7 @@ def finalize_response(self, request, response, *args, **kwargs):
         response = super(APIView, self).finalize_response(request, response, *args, **kwargs)
         time_started = getattr(self, 'time_started', None)
         response['X-API-Product-Version'] = get_awx_version()
-        response['X-API-Product-Name'] = 'AWX' if isinstance(get_licenser(), OpenLicense) else 'Red Hat Ansible Tower'
```
Comment (AlanCoding) [inline awx/api/generics.py:229]: > Red Hat Ansible Tower

since I'm consolidating code anyway...

File: awx/main/models/inventory.py [hunk 3]
```diff
@@ -1526,12 +1533,17 @@ class rhv(PluginFileInjector):
...
+    downstream_collection = 'rhv'
 
 
 class satellite6(PluginFileInjector):
     plugin_name = 'foreman'
     namespace = 'theforeman'
     collection = 'foreman'
+    downstream_namespace = 'redhat'
+    downstream_collection = 'satellite'
+    use_fqcn = True
```
Comment (AlanCoding) [inline awx/main/models/inventory.py:1530]: I don't love that I had to add this new property for this _one_ case where the upstream required the FQCN.... but that's what it does.

https://github.com/theforeman/foreman-ansible-modules/blob/develop/plugins/inventory/foreman.py#L28

**Comment by AlanCoding** [comment]
recheck

**Comment by AlanCoding** [comment]
That "awx-detect-schema-change :" failure looks like junk

**Comment by AlanCoding** [comment]
### Foreman / sat

I tested foreman (sat6) locally, and that redirected to the correct collection. That one is looking good. (there is a good deal of setup required to try this)

### ovirt / rhv

Next, I tested rhv

```
[WARNING]:  * Failed to parse /runner/inventory/ovirt.yml with auto plugin:
Invalid value "redhat.rhv.ovirt" for configuration option "plugin_type:
inventory plugin: ansible_collections.redhat.rhv.plugins.inventory.ovirt
setting: plugin ", valid values are: ['ovirt']
  File "/usr/lib/python3.8/site-packages/ansible/inventory/manager.py", line 290, in parse_source
    plugin.parse(self._inventory, self._loader, source, cache=cache)
  File "/usr/lib/python3.8/site-packages/ansible/plugins/inventory/auto.py", line 58, in parse
    plugin.parse(inventory, loader, path, cache=cache)
  File "/usr/share/ansible/collections/ansible_collections/redhat/rhv/plugins/inventory/ovirt.py", line 238, in parse
    config = self._read_config_data(path)
  File "/usr/lib/python3.8/site-packages/ansible/plugins/inventory/__init__.py", line 239, in _read_config_data
    self.set_options(direct=config, var_options=self._vars)
  File "/usr/lib/python3.8/site-packages/ansible/plugins/__init__.py", line 75, in set_options
    self._options = C.config.get_plugin_options(get_plugin_class(self), self._load_name, keys=task_keys, variables=var_options, direct=direct)
  File "/usr/lib/python3.8/site-packages/ansible/config/manager.py", line 365, in get_plugin_options
    options[option] = self.get_config_value(option, plugin_type=plugin_type, plugin_name=name, keys=keys, variables=variables, direct=direct)
```

Is this a bug? probably, I need to dive into the distributed collection to tell. But I'm guessing it is.

### tower / controller

As of yet, this is un-testable so I won't even try.

**Comment by AlanCoding** [comment]
Link https://github.com/oVirt/ovirt-ansible-collection/pull/293

**Comment by AlanCoding** [comment]
### Insights

```
[WARNING]:  * Failed to parse /runner/inventory/insights.yml with auto plugin:
inventory config '/runner/inventory/insights.yml' specifies unknown plugin
'redhatinsights.insights.insights'
  File "/usr/local/lib/python3.8/site-packages/ansible/inventory/manager.py", line 290, in parse_source
    plugin.parse(self._inventory, self._loader, source, cache=cache)
  File "/usr/local/lib/python3.8/site-packages/ansible/plugins/inventory/auto.py", line 53, in parse
    raise AnsibleParserError("inventory config '{0}' specifies unknown plugin '{1}'".format(path, plugin_name))
```

I suspect it's just too early for this one @wenottingham, the change was made to awx-ee, but it needs to make it to the downstream image.

**Review by sarabrajsingh — APPROVED** [review]

**Review by kdelee — APPROVED** [review]

**Pull request merged**